test/appium/tests/critical/test_wallet.py

```diff
@@ -0,0 +1,190 @@
+                                      eth_amount_receiver)
+
+
+@pytest.mark.xdist_group(name="new_one_2")
+@marks.new_ui_critical
+class TestWalletOneDevice(MultipleSharedDeviceTestCase):
+
+    def prepare_devices(self):
+        self.network_api = NetworkApi()
+        self.drivers, self.loop = create_shared_drivers(1)
+        self.sign_in_view = SignInView(self.drivers[0])
+        self.sign_in_view.create_user()
+        self.home_view = self.sign_in_view.get_home_view()
+        self.wallet_view = self.home_view.wallet_tab.click()
+
+    @marks.testrail_id(727231)
+    def test_wallet_add_remove_regular_account(self):
+        self.wallet_view.just_fyi("Adding new regular account")
+        new_account_name = "New Account"
+        self.wallet_view.add_regular_account(account_name=new_account_name)
+
+        if self.wallet_view.account_name_text.text != new_account_name:
+            pytest.fail("New account is not created")
+        self.wallet_view.account_emoji_button.click_until_presence_of_element(self.wallet_view.copy_address_button)
+        self.wallet_view.share_address_button.click()
+        new_wallet_address = self.wallet_view.sharing_text_native.text
+        self.wallet_view.click_system_back_button()
+        self.wallet_view.close_account_button.click_until_presence_of_element(self.home_view.show_qr_code_button)
+
+        self.wallet_view.just_fyi("Checking that the new wallet is added to the Sare QR Code menu")
+        self.home_view.show_qr_code_button.click()
+        self.home_view.share_wallet_tab_button.click()
+        if self.home_view.account_name_text.text != 'Account 1':
+            self.errors.append("Incorrect first account is shown on Share QR Code menu")
+        self.home_view.qr_code_image_element.swipe_left_on_element()
+        try:
+            self.home_view.account_name_text.wait_for_element_text(text=new_account_name, wait_time=3)
+            if self.home_view.copy_wallet_address() != new_wallet_address.split(':')[-1]:
+                self.home_view.driver.fail("Incorrect address")
+        except Failed:
+            self.errors.append("Can't swipe between accounts, newly added account is not shown")
+        self.home_view.click_system_back_button()
+
+        self.wallet_view.just_fyi("Removing newly added account")
+        if self.wallet_view.get_account_element(account_name=new_account_name).is_element_displayed():
+            self.wallet_view.remove_account(account_name=new_account_name)
+            if self.wallet_view.get_account_element(account_name=new_account_name).is_element_displayed():
+                self.errors.append("Account was not removed from wallet")
+        else:
+            self.errors.append("Newly added account is not shown in the accounts list")
+
+        self.errors.verify_no_errors()
+
+    @marks.testrail_id(727232)
+    def test_wallet_add_remove_watch_only_account(self):
```

Comment:
may be we can add some interesting account here and check collectibles, some unusual assets etc.

Can be done later I believe

Reply:
Totally agree, but it seems that not all the functionality is ready yet.